students/MichaelM/lesson9/mailroom_classes.py

```diff
@@ -0,0 +1,116 @@
+import os
+
+class Mailroom(object):
+    esteemed_donors_headers = ["Donor Name", "Total Donation", "Number of Donations", "Ave Donation"]
+    esteemed_donors_dict = {"john jacob astor": [10.0, 100.0, 1000.0, 10000.0],
+                            "alan rufus, 1st lord of richmond": [10.0],
+                            "henry ford": [10.0, 100.0, 1000.0],
+                            "cornelius vanderbilt": [10.0],
+                            "jakob fugger": [10.0, 100.0]}
+
+    def __init__(self):
+        pass
+
+
+    def donor_letters(letter_type="existing_donor", donor="", donation_amt=0.0, total_amt_donated=0.0):
+        total_donation = "${0:,.2f}".format(int(total_amt_donated))
+        letter = ""
+        if letter_type == "all_donors":
+            letter = f"{donor}, warm seasons greetings,\n" \
+                      f"Once again our society looks back upon this year's generousity from our many donors.\n" \
+                      f"Our records show to date your generosity of {total_donation} has been among the most generous." \
+                      f"We hope to continue our relationship into the next year and beyond.\n" \
+                      f"\n" \
+                      f"Kind regards,\n" \
+                      f"The Minimalist Society"
+        elif letter_type == "new":
+            letter = f"[Subject]: Thank you for your generous donation\n" \
+                      f"\n" \
+                      f"[Body]: {donor},\n" \
+                      f"Thank you for your generous donation. We are always grateful to greet new members of " \
+                      f"our ever expanding family.\n" \
+                      f"Your current and future beneficent contributions will go towards " \
+                      f"administering our clients with our services.\n" \
+                      f"Your initial contribution of {donation_amt} illustrates an exceedingly benevolent nature.\n" \
+                      f"\n" \
+                      f"Kind regards,\n" \
+                      f"The Minimalist Society\n" \
+                      f"[End]"
+        elif letter_type == "existing_donor":
+            letter = f"[Subject]: Once again, thank you for your generous donation\n" \
+                      f"\n" \
+                      f"[Body]: {donor},\n" \
+                      f"Thank you for your generous donation. Your beneficent contribution of {donation_amt} will go " \
+                      f"towards administering our clients with our services.\n" \
+                      f"Our records show to date your generosity " \
+                      f"of {total_donation} illustrates an exceedingly benevolent nature.\n" \
+                      f"\n" \
+                      f"Kind regards,\n" \
+                      f"The Minimalist Society\n" \
+                      f"[End]"
+        return letter
+
+class Donor(Mailroom):
```

Comment:
what is relationship between `Donor` and `Mailroom` classes?

Donor class should have no reference to multiple donors.